src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java

```diff
@@ -65,10 +65,14 @@ private List<DataField> getFields(ResourceEdge edge, ControlFieldsBuilder cfb) {
 
   private List<DataField> getFields(ControlFieldsBuilder cfb, ResourceEdge edge) {
     var resource = edge.getTarget();
-    var dataFields = rules.stream()
+    var fieldsFromResource = rules.stream()
       .flatMap(tagToRule -> mapToDataFields(edge, cfb, tagToRule));
-    return Stream.concat(dataFields, getOutgoingEdges(cfb, resource))
-      .toList();
+    var fieldsFromEdges = getOutgoingEdges(cfb, resource);
+    var combinedFields = Stream.concat(fieldsFromResource, fieldsFromEdges).toList();
+
+    return dataFieldPostProcessorSupplier.get()
```

Comment:
Here, we are working on a graph resource and its outgoing edges. Looking at the type of the current resource & the MARC tag, we can determine whether the MARC records has to be combined. So, moved the combiner logic to here.

We need to get a new `dataFieldPostProcessor` for each resource of the graph. Hence, using a `supplier` here, which will supply a new `dataFieldPostProcessor` for the current resource.